File: src/main/java/se/gewalli/commands/AddProductCommand.java
package se.gewalli.commands;

import com.fasterxml.jackson.annotation.*;
import se.gewalli.data.Repository;
import se.gewalli.entities.Product;

public record AddProductCommand(@JsonProperty("id") int id,
                                @JsonProperty("version") int version,
                                @JsonProperty("cost") float cost,
                                @JsonProperty("name") String name)  implements Command {

    @Override
    public CommandType getType() {
        return CommandType.AddProductCommand;
    }

    @Override
    public void run(Repository repository) {
        repository.save(new Product(id, cost, name, version));
    }
}

File: src/main/java/se/gewalli/commands/AddProductCommand.kt
package se.gewalli.commands

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import se.gewalli.data.Repository
import se.gewalli.entities.Product

class AddProductCommand @JsonCreator constructor(
    @JsonProperty("id") id: Int,
    @JsonProperty("version") version: Int,
    @param:JsonProperty("cost") val cost: Float,
    @param:JsonProperty("name") val name: String
) : Command(id, version) {
    override val type: CommandType get()= CommandType.AddProductCommand;

    override fun run(repository: Repository) {
        repository.save(Product(id, cost, name, version))
    }
}